sklearn/linear_model/tests/test_logistic.py

```diff
@@ -875,3 +872,53 @@ def test_warm_start():
                         assert_greater(2.0, cum_diff, msg)
                     else:
                         assert_greater(cum_diff, 2.0, msg)
+
+
+def test_logistic_slicing_weirdness():
+
+    digits = load_digits()
+    X, y = digits.data, digits.target
+    X = StandardScaler().fit_transform(X)
+    data = np.hstack((np.reshape(y.astype(float), (y.size, 1)), X))
+
+    X = data[:, 1:]
+    y = data[:, 0]
+
+    # classify small against large digits
+    y = (y > 4).astype(np.int)
```

Comment:
Better take the habit of never using Python type-based implicitly defined dtype (np.float and np.int) but instead `np.float64` and `np.int64`.